src/app/navigation/utils.ts

```diff
@@ -19,7 +19,7 @@ export const locationChangeForRoute = <R extends AnyRoute>(
 ): locationChange is Required<LocationChange<Match<R>>> =>
   !!locationChange.match && locationChange.match.route.name === route.name;
 
-export const getUrlRegexParams = <T extends {}>(obj: T): T => flatten(obj, {delimiter});
+export const getUrlRegexParams = <T extends {}>(obj: T): {} => flatten(obj, {delimiter});
```

Comment:
I think it was wrongly typed. It implied that flattened object will be the same as the original one. I changed it to `object` now since I've read that `{}` also allows primitives